StoreKitUnitTests/TrialOrIntroPriceEligibilityCheckerSK2Tests.swift

```diff
@@ -131,4 +133,122 @@ class TrialOrIntroPriceEligibilityCheckerSK2Tests: StoreKitConfigTestCase {
+                continuation.resume(returning: status)
+            }
+        })
+
+        expect(completionCalled) == true
+        expect(status) == .eligible
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testCheckEligibilityForLifetimeProductIsNoIntroOfferExists() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let productIdentifiers = Set([
+            "lifetime"
+        ])
+
+        let sk2Product = try await ProductsFetcherSK2().products(identifiers: productIdentifiers).first
+        let receivedProduct = try XCTUnwrap(sk2Product)
+        let storeProduct = StoreProduct.from(product: receivedProduct)
+
+        var completionCalled = false
+
+        let status = try await withCheckedThrowingContinuation({ (continuation: ContinuationStatusResult) in
+            self.trialOrIntroPriceEligibilityChecker!.checkEligibility(product: storeProduct) { status in
+                completionCalled = true
+                continuation.resume(returning: status)
+            }
+        })
+
+        expect(completionCalled) == true
+        expect(status) == .noIntroOfferExists
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testCheckEligibilityForProductIsIneligibleAfterPurchasing() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let productIdentifiers = Set([
+            "com.revenuecat.monthly_4.99.1_week_intro"
+        ])
+
+        let sk2Product = try await ProductsFetcherSK2().products(identifiers: productIdentifiers).first
+        let receivedProduct = try XCTUnwrap(sk2Product)
+        let storeProduct = StoreProduct.from(product: receivedProduct)
+
+        var completionCalled = false
+
+        let prePurchaseStatus = try await withCheckedThrowingContinuation({ (continuation: ContinuationStatusResult) in
+            self.trialOrIntroPriceEligibilityChecker!.checkEligibility(product: storeProduct) { status in
+                completionCalled = true
+                continuation.resume(returning: status)
+            }
+        })
+
+        expect(completionCalled) == true
+        expect(prePurchaseStatus) == .eligible
+
+        let purchasableSK2Product = try XCTUnwrap(storeProduct.sk2Product)
+        _ = try await purchasableSK2Product.purchase()
+
+        completionCalled = false
+
+        let postPurchaseStatus = try await withCheckedThrowingContinuation({ (continuation: ContinuationStatusResult) in
+            self.trialOrIntroPriceEligibilityChecker!.checkEligibility(product: storeProduct) { status in
+                completionCalled = true
+                continuation.resume(returning: status)
+            }
+        })
+
+        expect(completionCalled) == true
+        expect(postPurchaseStatus) == .ineligible
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testCheckEligibilityForProductIsUnknown() async throws {
```

Comment:
perhaps something like this is more clear? 
```suggestion
    func testCheckEligibilityForInvalidProductIsUnknown() async throws {
```